aya/src/programs/mod.rs

```diff
@@ -971,3 +972,43 @@ impl ProgramInfo {
         Ok(ProgramInfo(info))
     }
 }
+
+/// Programs is an Iterator over loaded eBPF programs
+pub struct Programs {
+    current: u32,
+}
+
+impl Iterator for Programs {
+    type Item = Result<ProgramInfo, ProgramError>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let current = self.current;
+
+        match bpf_prog_get_next_id(current) {
+            Some(next) => {
+                self.current = next;
+                Some(
+                    bpf_prog_get_fd_by_id(next)
+                        .map_err(|io_error| ProgramError::SyscallError {
+                            call: "bpf_prog_get_fd_by_id".to_owned(),
+                            io_error,
+                        })
+                        .and_then(|fd| {
+                            bpf_prog_get_info_by_fd(fd)
+                                .map_err(|io_error| ProgramError::SyscallError {
+                                    call: "bpf_prog_get_info_by_fd".to_owned(),
+                                    io_error,
+                                })
+                                .map(ProgramInfo)
+                        }),
+                )
+            }
+            None => None,
+        }
+    }
+}
+
+/// Returns an iterator over all loaded bpf programs
+pub fn programs() -> Programs {
```

Comment:
I think this should be called `all_loaded_programs` or something to differentiate it from `Bpf::programs`, and the documentation should explain the difference between the two

Reply:
I was about to ask you about that! I was thinking maybe re-export this fn at the top level of the crate and call it `iter_programs()`. Better docs + an example/doctest is obviously required - i'll get on that now 😆